yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -973,38 +985,59 @@ func areMinVersionsFixedInEqual(m1 map[string]*ybversion.YBVersion, m2 map[strin
+func getUnsupportedFeaturesFromSchemaAnalysisReport(category string, featureName string, issueReason string, issueType string, schemaAnalysisReport utils.SchemaReport, displayDDLInHTML bool, description string) UnsupportedFeature {
 	log.Info("filtering issues for feature: ", featureName)
 	objects := make([]ObjectInfo, 0)
 	link := "" // for oracle we shouldn't display any line for links
 	var minVersionsFixedIn map[string]*ybversion.YBVersion
 	var minVersionsFixedInSet bool
 
-	for _, issue := range schemaAnalysisReport.Issues {
-		if !slices.Contains([]string{UNSUPPORTED_FEATURES, MIGRATION_CAVEATS}, issue.IssueType) {
+	for _, analyzeIssue := range schemaAnalysisReport.Issues {
+		if !slices.Contains([]string{UNSUPPORTED_FEATURES, MIGRATION_CAVEATS}, analyzeIssue.IssueType) {
 			continue
 		}
-		issueMatched := lo.Ternary[bool](issueType != "", issueType == issue.Type, strings.Contains(issue.Reason, issueReason))
+		issueMatched := lo.Ternary[bool](issueType != "", issueType == analyzeIssue.Type, strings.Contains(analyzeIssue.Reason, issueReason))
 
 		if issueMatched {
-			objectInfo := ObjectInfo{
-				ObjectName:   issue.ObjectName,
-				SqlStatement: issue.SqlStatement,
-			}
-			link = issue.DocsLink
-			objects = append(objects, objectInfo)
 			if !minVersionsFixedInSet {
-				minVersionsFixedIn = issue.MinimumVersionsFixedIn
+				minVersionsFixedIn = analyzeIssue.MinimumVersionsFixedIn
 				minVersionsFixedInSet = true
 			}
-			if !areMinVersionsFixedInEqual(minVersionsFixedIn, issue.MinimumVersionsFixedIn) {
-				utils.ErrExit("Issues belonging to UnsupportedFeature %s have different minimum versions fixed in: %v, %v", featureName, minVersionsFixedIn, issue.MinimumVersionsFixedIn)
+			if !areMinVersionsFixedInEqual(minVersionsFixedIn, analyzeIssue.MinimumVersionsFixedIn) {
+				utils.ErrExit("Issues belonging to UnsupportedFeature %s have different minimum versions fixed in: %v, %v", featureName, minVersionsFixedIn, analyzeIssue.MinimumVersionsFixedIn)
+			}
+
+			objectInfo := ObjectInfo{
+				ObjectName:   analyzeIssue.ObjectName,
+				SqlStatement: analyzeIssue.SqlStatement,
 			}
+			link = analyzeIssue.DocsLink
+			objects = append(objects, objectInfo)
+
+			assessmentReport.AppendIssues(convertAnalyzeSchemaIssueToAssessmentIssue(analyzeIssue, category, description, minVersionsFixedIn))
 		}
 	}
+
 	return UnsupportedFeature{featureName, objects, displayDDLInHTML, link, description, minVersionsFixedIn}
 }
 
+// Q: do we no need of displayDDLInHTML in this approach? DDL can always be there for issues in the table if present.
+func convertAnalyzeSchemaIssueToAssessmentIssue(analyzeSchemaIssue utils.AnalyzeSchemaIssue, category string, issueDescription string, minVersionsFixedIn map[string]*ybversion.YBVersion) AssessmentIssue {
+	return AssessmentIssue{
```

Comment:
I'm deferring the review of these fields to when we get to the UI task. At that time, we can properly discuss this.

Reply:
yeah makes sense, i have also not spent a lot of time on this right now